services/replay/service.go

```diff
@@ -1215,6 +1229,16 @@ func (r *Service) doReplay(id string, task *kapacitor.Task, runReplay func(tm *k
 	if err != nil {
 		return errors.Wrap(err, "running replay")
 	}
+	stats, err := tm.ExecutionStats(task.ID)
+	if err != nil {
+		return errors.Wrap(err, "getting executing stats replay")
+	}
+
+	// Update replay
+	err = r.updateReplayResult(id, stats, err)
+	if err != nil {
+		return errors.Wrap(err, "update finished replay")
+	}
```

Comment:
I wonder if this is the best way to handle this error after the refactoring. 

The problem might be that if `runReplay` or ` tm.ExecutionStats(task.ID)` calls fail, `updateReplayResult` will never be called and the replay will end up with status Running instead of Failed.

I will refactor the call to r.updateReplayResult as a function and pass it in the previous errors

Reply:
Hmm, yeah this didn't work out very well, sorry about that.

Here is another idea, revert the updateReplayResult changes and just pass a pointer to the Replay to the doReplay function and have it set the stats on the Replay object if it finishes without error. Do not have the doReplay function actually store the replay, but leave that as part of the updateReplayResult function as it was before. This way updateReplayResult doesn't know or care if the stats were set. Does that seem like a better solution?